### PR TITLE
Fix incorrect pending reward calculation

### LINKED ISSUE #7369: Inaccurate Display of Pending Rewards on Mirror Node
### Description

The Blade team encountered an issue where the pending rewards on the mirror node consistently display as zero if the account has claimed or restaked since the 9th of December (if this hasn't happened, the pending reward will just be inaccurate). The discrepancy arises because the pending rewards, although present, are not reflected accurately in the user interface.

**Expected Behavior:**
The mirror node should accurately display the pending rewards at all times, providing users with correct and up-to-date information about their earnings.

**Actual Behavior:**
The pending rewards are shown as zero or are inaccurate, despite the fact that users are earning rewards. This issue leads to confusion and potential mistrust in the system.

### Steps to reproduce

1. Observe the rewards displayed on the mirror node for an account that has claimed or restaked since the 9th (if they haven't done this then the pending reward wont show 0 but wont be accurate either.)
2. Notice that the account shows zero rewards available for claiming. (supporting pic shows inaccurate reward)
3. Re-stake or transfer hbar to the account.
4. Observe that rewards are then distributed, contradicting the initial display of zero rewards.

### Additional context
Pic related (again pending reward is not accurate):
image

### Hedera network

mainnet

### Version

v0.93.1

### Operating system

None

### PR DESCRIPTION
**Pull request by xin-hedera**
**Description**:

This PR fixes the incorrect pending reward calculation bug.

- Use deduped `account_balance` info in `EntityStakeRepository.createEntityStateStart()`
- Add SQL migration to remove incorrect entity_stake rows

**Related issue(s)**:

Fixes #7369 

**Notes for reviewer**:

Still testing with mainnet-staging database, the pending rewards reconciliation is going to take some time. For mainnet-staging, data since 11/23/23 needs to be recalculated and one staking period generally takes about 8 minutes.

Update: pending reward is correct after reconciling 19 staking periods. However, it takes almost 45 minutes each for the last 4. I'll create a ticket with some solution to optimize it.

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-importer/src/main/resources/db/migration/v1/V1.91.0__remove_incorrect_entity_stake.sql [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  where timestamp_range @> migration_timestamp;
+
+  -- Delete incorrect history rows
+  raise notice 'deleting entity_stake_history rows at or after: %', last_good_lower_timestamp;
+  delete from entity_stake_history
+  where timestamp_range in (
+    select timestamp_range
+    from entity_stake_history
+    where id = 800 and lower(timestamp_range) >= last_good_lower_timestamp
+  );
```
Comment (xin-hedera) [inline hedera-mirror-importer/src/main/resources/db/migration/v1/V1.91.0__remove_incorrect_entity_stake.sql:48]: This is the most time consuming part of the function. For mainnet-staging, it has to delete all rows from 11/23/23. Overall the whole migration takes about 11 minutes.

**Review by jnels124 — APPROVED** [review]
LGTM

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/repository/EntityStakeRepository.java [hunk 2]
```diff
@@ -62,7 +62,9 @@ select lower(timestamp_range) as timestamp from entity_history where id = 800
         ), balance_timestamp as (
              select ab.consensus_timestamp
              from account_balance ab, end_period ep
-             where ab.account_id = 2 and ab.consensus_timestamp <= ep.consensus_timestamp
+             where ab.account_id = 2 and
+               ab.consensus_timestamp > (ep.consensus_timestamp - 2592000000000000) and
```
Comment (jascks) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/repository/EntityStakeRepository.java:66]: What is the significance of 2592000000000000?

Comment (edwin-greene) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/repository/EntityStakeRepository.java:66]: It is a constant already in use in `AccountBalanceRepository` and `TokenBalanceRepository`, representing a month in nanoseconds. It is used to ensure the query is getting the deduplicated balances, hitting at most 2 partitions.

Comment (jascks) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/repository/EntityStakeRepository.java:66]: Cool. Thank you!

Comment (xin-hedera) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/repository/EntityStakeRepository.java:66]: Just realized the value is 30 days not 31 days. I need to change this to 31 days, the max number of days in a month. It's not a problem to use 30 days to find the balance snapshot timestamp, however, it's a problem to use 30 days as a range to find deduped account balance info.

Comment (jnels124) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/repository/EntityStakeRepository.java:66]: maybe 3 partitions if requested in earyly days of march :)

Comment (steven-sheehy) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/repository/EntityStakeRepository.java]: Then isn't it wrong in AccountBalanceRepository and TokenBalanceRepository as well?

Comment (steven-sheehy) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/repository/EntityStakeRepository.java]: Nevermind, Xin just said it's not a problem in his last sentence. :D

**Review by jnels124 — APPROVED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]

**Review by jascks — APPROVED** [review]
LGTM

**Pull request merged**